pkg/controllers/applicationfailover/rb_application_failover_controller.go

```diff
@@ -157,8 +157,11 @@ func (c *RBApplicationFailoverController) evictBinding(binding *workv1alpha2.Res
 		switch binding.Spec.Failover.Application.PurgeMode {
 		case policyv1alpha1.Graciously:
 			if features.FeatureGate.Enabled(features.GracefulEviction) {
-				binding.Spec.GracefulEvictCluster(cluster, workv1alpha2.NewTaskOptions(workv1alpha2.WithProducer(RBApplicationFailoverControllerName),
-					workv1alpha2.WithReason(workv1alpha2.EvictionReasonApplicationFailure), workv1alpha2.WithGracePeriodSeconds(binding.Spec.Failover.Application.GracePeriodSeconds)))
+				binding.Spec.GracefulEvictCluster(cluster, workv1alpha2.NewTaskOptions(
```

Comment:
Eventually this can be combined into a single case statement with Graciously + Immediately, but while the FeatureGate exists, we'll need to separate them.

Reply:
Agree.

Besides, Do we also need to deal with Never's situation?

Reply:
> Besides, Do we also need to deal with Never's situation?

+1

Reply:
I see, so in the case of Never we can generate a GracefulEvictionTask (which won't actually evict the workload), but will be used to filter out the cluster that the application is failing over from?

What are the use cases for an application never being cleaned up? Just out of curiosity.

Reply:
> I see, so in the case of Never we can generate a GracefulEvictionTask (which won't actually evict the workload), but will be used to filter out the cluster that the application is failing over from?

Yes, it's like this.